nixos/modules/services/web-apps/wordpress.nix

```diff
@@ -106,62 +106,36 @@ let
         };
 
         plugins = mkOption {
-          type = types.listOf types.path;
-          default = [];
+          type = types.attrsOf types.path;
+          default = {};
           description = lib.mdDoc ''
-            List of path(s) to respective plugin(s) which are copied from the 'plugins' directory.
+            Path(s) to respective plugin(s) which are copied from the 'plugins' directory.
 
             ::: {.note}
             These plugins need to be packaged before use, see example.
             :::
           '';
           example = literalExpression ''
-            let
-              # Wordpress plugin 'embed-pdf-viewer' installation example
-              embedPdfViewerPlugin = pkgs.stdenv.mkDerivation {
```

Comment:
Is it good to remove the example on manually packaging and using plugins/themes? Right now it is only unofficially covered in the NixOS Wiki. Maybe we should add a note to the NixOS manual on how to package plugins and themes in the repository or how to manually package and use them in the Wordpress module?
This could be done in a separate PR tough

Reply:
I would be happy to adapt this to the new format if that is the consensus, however there are three ways I have seen this done, so I was tempted to not give any suggestions via the nixos module examples:
- official modules via `wp4nix` (involves some non-trivial derviation)
- fetchzip (see `pkgs.wordpressPackages.plugins.civicrm`)
- this approach